### ISSUE TITLE
Can't connect Paypal transactions with Open Collective expenses

### ISSUE THREAD

**Issue by alanna** [opencollective/opencollective#1149]
Recently we had an issue with a Paypal transaction. It was reversed and Paypal sent through a notification. But there seems to be no way to find the associated Open Collective expense. 

When I do manual Paypal payments, I always paste in the expense URL into the notes, but these notes are not displayed anywhere in the Paypal history. And for payments through the API there is no opportunity to include notes. This makes it very hard to follow up the issue and check details when there's an issue with the transaction.

Is there any way to search Open Collective with a Paypal transaction ID, or user email address? Or, if we had the ability to search Open Collective by expense amount and date range, it would be possible to narrow it down and find it manually.

**Comment by xdamman** [comment]
Yes we totally need a way to search / filter expenses (but also transactions).

Adding tags to them would also be helpful.

**Comment by alannallama** [comment]
We still need this

**Comment by alannallama** [comment]
Why was this closed? I thought if stalebot comments and you comment back it will stay open?

**Comment by alannallama** [comment]
We absolutely need this in the new expense design.

**Comment by alanna** [comment]
Just bumping this again because I ran into it again today. When someone refers to an expense and there has been a problem with Paypal, it's very very hard to connect the transactions on our platform to the paypal data to verify if a certain payment has failed or needs to be changed, etc.

**Comment by znarf** [comment]
Closing: this has been implemented, it just never got linked back here.

- Searching expenses by PayPal transaction ID (and Wise transfer/batch ID, virtual card transaction ID) landed in opencollective/opencollective-api#9268 (August 2023), which resolved the duplicate #6864. Paste the merchant transaction ID into the expense search in the host dashboard and it matches on the expense's payout data. The lookup is indexed.
- Searching expenses by payee email landed in opencollective/opencollective-api#9457 (November 2023).
- Amount and date range filters are available on the same expenses page.
